telegram/ext/_application.py

```diff
@@ -1019,14 +1033,15 @@ def __create_task_done_callback(self, task: asyncio.Task) -> None:
 
     async def __create_task_callback(
         self,
-        coroutine: Union[Generator[Optional["asyncio.Future[object]"], None, RT], Awaitable[RT]],
+        coroutine: _CoroType,
         update: Optional[object] = None,
         is_error_handler: bool = False,
     ) -> RT:
         try:
-            if isinstance(coroutine, Generator):
+            # Generator-based coroutines are not supported in Python 3.12+
+            if sys.version_info < (3, 12) and isinstance(coroutine, Generator):
```

Comment:
Can we issue a deprecation warning here?

Reply:
good idea, will try